app/Platform/Actions/BuildHubBreadcrumbsAction.php

```diff
@@ -0,0 +1,393 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Platform\Actions;
+
+use App\Models\GameSet;
+use App\Platform\Data\GameSetData;
+use App\Platform\Enums\GameSetType;
+use Illuminate\Support\Facades\Cache;
+
+/**
+ * Builds breadcrumb navigation paths for hub (game set) hierarchies.
+ *
+ * This action  generates breadcrumb navigation paths for hubs, handling various
+ * special cases and hierarchical relationships. It supports different types of
+ * game sets including themes, difficulty levels, and various hub types.
+ *
+ * - Caches breadcrumb paths with fresh/stale timing.
+ * - Handles special cases for Theme and Difficulty hubs.
+ * - Supports derived hierarchical navigation through hub types.
+ * - Manages hub relationships via predefined mappings.
+ */
+class BuildHubBreadcrumbsAction
```

Comment:
Overall, this seems to work really well, but I did find an oddity:
![image](https://github.com/user-attachments/assets/24580589-d16b-4c2e-a6ee-bd31283adc2a)

I think it should be "All Hubs > Miscellaneous > Virtual Console > New Nintendo 3DS"

Reply:
And `Publisher` isn't being extracted from:
![image](https://github.com/user-attachments/assets/9ac7b4e3-aac2-4e64-b415-bc43e55ca9b6)

Reply:
And a couple more oddities:
![image](https://github.com/user-attachments/assets/da6451a9-f523-4790-8dff-caaa87dcb656)
![image](https://github.com/user-attachments/assets/1e3d534c-0e75-408d-867b-a80c879fcb39)
![image](https://github.com/user-attachments/assets/72cd0396-2b15-456a-9cad-148307e91b5e)
![image](https://github.com/user-attachments/assets/cc1eb58b-6cea-4488-9c2f-6e7210ba76e9)

Reply:
I've updated the action's logic in latest to hopefully handle these cases.

I was unable to repro the "Publisher - Absolute Entertainment" case - that seemed to have been working correctly on my local.

Reply:
"Publisher - Absolute Entertainment" is failing for me because the hub isn't linked to "Central - Publisher". That appears to have been done more recently than my dump
![image](https://github.com/user-attachments/assets/36456d08-cef1-4466-b75d-1ff478528003)

Which brings up the fact that there's no way to add that association in this state. I'm not sure it's reasonable to merge this without
 >PR 2: Create management tools for hubs
 
 I guess that's what the feature flag is for?

Reply:
> I guess that's what the feature flag is for?

Yes, while the feature flag is disabled (ie: prod), users should only be able to navigate to these pages by landing on them directly. I agree we can't make this generally available without management tools. I didn't want to squeeze that functionality into this PR because there's already so much stuff in this branch.

Reply:
> Unwanted - Current Games and Unwanted - Past Games are still parenting to each other. They should parent to Events - Unwanted.

This should now be fixed in latest, though it should be noted I've removed the ⭐️ characters from the hub titles in both my local environment and prod:

<img width="563" alt="Screenshot 2024-12-12 at 3 09 24 PM" src="https://github.com/user-attachments/assets/40b61b82-704b-4038-aced-4a1dd2c1d834" />